File: src/CSP-AF/CSPAgents/CSPUtilityServerGroups/CSPVerifyAgentIsRunning.cs
using System;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Models;
using CloudSuitePortal.Services;
using CloudSuitePortal.Controllers;

namespace actions.CSP.CSPAgents.CSPUtilityServerGroups
{
    class CSPVerifyAgentIsRunning
    {
        public void run(Dictionary<string, object> Params)
        {
            var guid = (string)Params["GUID"];
            var navigationController = CSPBrowser.Application.WindowController.NavigationController;
            var controller = (CSPUtilityServerGroupsController)navigationController.VisibleViewController;
            controller.VerifyAgentIsRunning(guid);
        }
    }
}

File: src/CSPAF/CSPAgents/CSPUtilityServerGroups/CSPVerifyAgentIsRunning.java
package CSPAF.CSPAgents.CSPUtilityServerGroups;

import java.io.*;
import java.util.*;

import com.infor.cloudsuite.portal.core.services.*;
import com.infor.cloudsuite.portal.core.controllers.*;
import com.infor.cloudsuite.portal.singletenant.agents.controllers.*;

class CSPVerifyAgentIsRunning
{
    public void run(HashMap<String, Object> params) throws IOException, InterruptedException
    {
        CSPNavigationController navigationController = CSPBrowser.getApplication().getWindowController().getNavigationController();
        CSPUtilityServerGroupsController controller = (CSPUtilityServerGroupsController)navigationController.getVisibleViewController();
        controller.verifyAgentIsRunning();
    }
}